zokrates_core/src/helpers/libsnark_gadget.rs

```diff
@@ -28,6 +30,15 @@ impl<T: Field> Executable<T> for LibsnarkGadgetHelper {
 					return Err(format!("{}", e));
 				}
 
+				Ok(witness_result.unwrap().variables.iter().map(|&i| T::from(i)).collect())
```

Comment:
Instead of duplicating the whole case, could we have only the minimum in there and the rest after the match as it's common?